gemini/sample-apps/gemini-mesop-cloudrun/main.py

```diff
@@ -0,0 +1,1650 @@
+    HarmBlockThreshold,
+    HarmCategory,
+    Part,
+)
+
+from shared.navmenu import navmenu
+
+
+PROJECT_ID = os.environ.get("GOOGLE_CLOUD_PROJECT")  # Your Google Cloud Project ID
+LOCATION = os.environ.get("GOOGLE_CLOUD_REGION")  # Your Google Cloud Project Region
+vertexai.init(project=PROJECT_ID, location=LOCATION)
+
+
+@dataclass_json
+@me.stateclass
+class State():
+    model: str = "gemini-1.5-flash-002"
+    current_page: str = "/"
+
+    # Story
+    input: str = ""
+    story_character_name: str
+    story_character_type: str = "Cat"
+    story_character_personality: str = "Mitten is a very friendly cat."
+    story_character_location: str = "Andromeda Galaxy"
+    story_selected_premises: list[str] = field(default_factory=lambda: ["adventure"])
+    story_temp_value: str = "low"
+    story_length_value: str = "short"
+    story_progress: bool = False
+    story_output: str = ""
+
+    # Marketing
+    marketing_product: str = "ZomZoo"
+    marketing_product_categories: list[str] = field(default_factory=lambda: ["clothing", "electronics", "food", "health & beauty", "home & garden"])
+    marketing_product_category: str = "clothing"
+    marketing_target_audiences: list[str] = field(default_factory=lambda: ["18-24", "25-34", "35-44", "45-54", "55-64", "65+"])
+    marketing_target_audience: str = "18-24"
+    marketing_target_locations: list[str] = field(default_factory=lambda: ["urban","suburban","rural"])
+    marketing_target_location: str = "urban"
+    marketing_campaign_goals: list[str] = field(default_factory=lambda: ["increase brand awareness", "generate leads", "drive sales", "improve brand sentiment"])
+    marketing_campaign_goal: str = "increase brand awareness"
+    marketing_campaign_selected_goals: list[str] = field(default_factory=lambda: ["increase brand awareness", "generate leads"])
+    marketing_brand_voices: list[str] = field(default_factory=lambda: ["formal","informal", "serious", "humorous", "casual"])
+    marketing_brand_voice: str = "formal"
+    marketing_budgets: list[str] = field(default_factory=lambda: ["1,000 - 5,000", "5,000 - 10,000", "10,000 - 20,000", "20,000+" ])
+    marketing_budget: str = "1,000 - 5,000"
+    marketing_campaign_progress: bool = False
+    marketing_campaign_output: str = ""
+
+    # Image playground
+    image_tab: str = "furniture"
+
+    furniture_recommendation_output: str = ""
+    oven_instructions_output: str = ""
+    er_doc_output: str = ""
+    image_glasses_shape_radio_value: str = "oval"
+    image_glasses_output_radio_value: str = "text"
+    glasses_rec_output: str = ""
+    math_answers_output: str = ""
+
+    image_progress_spinner: bool = False
+
+    # Video playground
+    video_tab: str = "desc"
+
+    video_spinner_progress: bool = False
+    video_description_content: str = ""
+    video_tags_content: str = ""
+    video_highlights_content: str = ""
+    video_geolocation_content: str = ""
+
+
+# Helpers
+def gcsurl_to_httpurl(gcsuri: str) -> str:
```

Comment:
```suggestion
def gcs_to_http(gcsuri: str) -> str:
```
To satisfy spell-check and make it more compact.